pkg/loadbalancer/reflectors/conversions.go

```diff
@@ -0,0 +1,386 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reflectors
+
+import (
+	"log/slog"
+	"net"
+	"slices"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/cilium/cilium/pkg/annotation"
+	"github.com/cilium/cilium/pkg/cidr"
+	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/ip"
+	"github.com/cilium/cilium/pkg/k8s"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/source"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+var (
+	zeroV4 = cmtypes.MustParseAddrCluster("0.0.0.0")
+	zeroV6 = cmtypes.MustParseAddrCluster("::")
+
+	ingressDummyAddress = cmtypes.MustParseAddrCluster("192.192.192.192")
+	ingressDummyPort    = uint16(9999)
+)
+
+func isIngressDummyEndpoint(l3n4Addr loadbalancer.L3n4Addr) bool {
+	// The ingress and gateway-api controllers (operator/pkg/model/translation/{gateway-api,ingress}) create
+	// a dummy endpoint to force Cilium to reconcile the service. This is no longer required with this new
+	// control-plane, but due to rolling upgrades we cannot remove it immediately. Hence we have the
+	// special handlind here to just ignore this endpoint to avoid populating the tables with unnecessary
+	// data.
+	return l3n4Addr.AddrCluster.Equal(ingressDummyAddress) && l3n4Addr.Port == ingressDummyPort
+}
+
+func isHeadless(svc *slim_corev1.Service) bool {
+	_, headless := svc.Labels[corev1.IsHeadlessService]
+	if strings.ToLower(svc.Spec.ClusterIP) == "none" {
+		headless = true
+	}
+	return headless
+}
+
+func convertService(cfg loadbalancer.ExternalConfig, log *slog.Logger, svc *slim_corev1.Service, source source.Source) (s *loadbalancer.Service, fes []loadbalancer.FrontendParams) {
```

Comment:
Wondering if a skipped entry should be logged at all or not (as it is now)? Like I am setting up a ipv6 service, but cfg.EnabledIPv6 is not enabled. wdyt?

Reply:
This is used by Cilium in general so starting to suddenly log these is definitely too aggressive. However, we could debug log it. Perhaps makes sense to add debug logs for all the potentially surprising skips.

Reply:
Added debug logs.